### ISSUE TITLE
Unhandled Exception: MissingPluginException

### ISSUE THREAD

**Issue by circledata** [Dev-hwang/flutter_foreground_task#49]
I'm trying to using this plugin with flutter_beacon but i keep getting the error below:

**Unhandled Exception: MissingPluginException(No implementation found for method initialize on channel flutter_beacon)**

I suppose it has to do with the fact that flutter_foreground_task launches a new isolate. Is there a way around this? Really stuck.

**Comment by Dev-hwang** [comment]
@circledata 

**- Please note that the description below may not be accurate.**

The `flutter_foreground_task` plugin creates a new isolate(Background thread) separate from the main isolate(UI thread). When the app runs on the Android platform, the screen is drawn with the UI thread. The object corresponding to the screen is an [Activity](https://developer.android.com/guide/components/activities/intro-activities). Compared to this, new isolate is a thread for background processing and does not directly draw or create the screen.

Look at the native code of the [flutter_beacon](https://github.com/alann-maulana/flutter_beacon/blob/master/android/src/main/java/com/flutterbeacon/FlutterBeaconPlugin.java) library, the `onAttachedToActivity` function is initializing the channel(MethodChannel, EventChannel). This function is called when an Activity is attached to the Flutter Engine. 

Since `flutter_foreground_task` plugin only creates new isolate, Activity object is not created and `onAttachedToActivity` function is not called. This seems to be causing the MissingPluginException.

In conclusion, the `flutter_foreground_task` plugin and the `flutter_beacon` plugin cannot be used together. If you want to use it together, the channel must be initialized in the `onAttachedToEngine` function like this [plugin](https://github.com/Dev-hwang/flutter_location/blob/master/fl_location/android/src/main/kotlin/com/pravera/fl_location/FlLocationPlugin.kt).

**Comment by Dev-hwang** [comment]
@circledata 

**- Please note that the description below may not be accurate.**

The `flutter_foreground_task` plugin creates a new isolate(Background thread) separate from the main isolate(UI thread). When the app runs on the Android platform, the screen is drawn with the UI thread. The object corresponding to the screen is an [Activity](https://developer.android.com/guide/components/activities/intro-activities). Compared to this, new isolate is a thread for background processing and does not directly draw or create the screen.

Look at the native code of the [flutter_beacon](https://github.com/alann-maulana/flutter_beacon/blob/master/android/src/main/java/com/flutterbeacon/FlutterBeaconPlugin.java) library, the `onAttachedToActivity` function is initializing the channel(MethodChannel, EventChannel). This function is called when an Activity is attached to the Flutter Engine. Since `flutter_foreground_task` plugin only creates new isolate, Activity object is not created and `onAttachedToActivity` function is not called. This seems to be causing the MissingPluginException.

In conclusion, the `flutter_foreground_task` plugin and the `flutter_beacon` plugin cannot be used together. If you want to use it together, the channel must be initialized in the `onAttachedToEngine` function like this [plugin](https://github.com/Dev-hwang/flutter_location/blob/master/fl_location/android/src/main/kotlin/com/pravera/fl_location/FlLocationPlugin.kt).
